src/aggregations/top-hits-aggregation.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Internal counter for added top hits aggregations.
+ * @type {Number}
+ */
+let count = 0;
+
+/**
+ * Construct a Top hits aggregation.
+ *
+ * @param  {Object} opts  Options to include in the aggregation (maybe empty).
+ * @param  {String} name  Aggregation name. Defaults to agg_top_hits_{count}.
+ * @return {Object}       Top hits Aggregation.
+ */
+export default function topHitsAggregation(opts, name) {
+  name = name || `agg_top_hits_${count++}`
```

Comment:
Ah, this is an interesting side effect and consequence of elasticsearch not liking multiple aggregations with the same name, right?